src/pyodide/internal/loadPackage.ts

```diff
@@ -140,8 +140,19 @@ async function loadPackagesImpl(
   Module.FS.mount(tarFS, { info }, path);
 }
 
+/**
+ * Downloads the requirements specified and loads them into Pyodide. Note that this does not
+ * do any dependency resolution, it just installs the requirements that are specified. See
+ * `getTransitiveRequirements` for the code that deals with this.
+ */
 export async function loadPackages(Module: Module, requirements: Set<string>) {
-  const pkgsToLoad = requirements.union(new Set(STDLIB_PACKAGES));
+  let pkgsToLoad = requirements;
+  // TODO: Package snapshot created with '20240829.4' needs the stdlib packages to be added here.
+  // We should remove this check once the next Python and packages versions are rolled
+  // out.
+  if (PACKAGES_VERSION == '20240829.4') {
```

Comment:
Let's add to `metadata.ts`:
```
export const USING_OLDEST_PACKAGES_VERSION = PACKAGES_VERSION === '20240829.4';
```
and then use that in the three places we have this comparison.